src/ComputeSharp.D2D1/Shaders/Interop/D2D1ShaderCompiler.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Buffers;
+using System.Text;
+using ComputeSharp.D2D1.Exceptions;
+#if !NET6_0_OR_GREATER
+using ComputeSharp.D2D1.NetStandard.System.Text;
+#endif
+using ComputeSharp.D2D1.Shaders.Translation;
+using TerraFX.Interop.DirectX;
+using TerraFX.Interop.Windows;
+
+namespace ComputeSharp.D2D1.Interop;
+
+/// <summary>
+/// Provides methods to manually compile D2D1 shaders.
+/// </summary>
+public static class D2D1ShaderCompiler
+{
+    /// <summary>
+    /// Compiles a new HLSL shader from the input source code.
+    /// </summary>
+    /// <param name="source">The HLSL source code to compile.</param>
+    /// <param name="entryPoint">The entry point of the shader being compiled.</param>
+    /// <param name="shaderProfile">The shader profile to use to compile the shader.</param>
+    /// <param name="options">The compiler options to use to compile the shader.</param>
+    /// <returns>The bytecode for the compiled shader.</returns>
+    /// <exception cref="FxcCompilationException">Thrown if the compilation fails.</exception>
+    public static ReadOnlyMemory<byte> Compile(
+        ReadOnlySpan<char> source,
```

Comment:
consider naming this `hlslSource`

Reply:
Done in fe49c2870af7a2934f0d45e59159c02ea68e4eaa 🙂